File: examples/cpl_init/cfd_init.f90
program main_CFD
   use cpl, only : CPL_init, CPL_finalize
   use mpi
   implicit none

   integer :: rank, nprocs, ierr
   integer :: CFD_COMM
   integer, parameter :: CFD_realm=1

   !Initialise MPI
   call MPI_Init(ierr)

   !Create MD Comm by spliting world
   call CPL_init(CFD_realm, CFD_COMM, ierr)

   !get local processor rank and print
   call MPI_comm_size(CFD_COMM, nprocs, ierr)
   call MPI_comm_rank(CFD_COMM, rank, ierr)

   print*, "CFD code processor ", rank+1, " of ", nprocs

   !No need for seperate CPL finalise as MPI finalise takes care of this
   call CPL_finalize(ierr)
   call MPI_finalize(ierr)

end program main_CFD

File: examples/cpl_init/cfd_init.cpp
#include "CPL.h"
#include "mpi.h"
#include <iostream>

using namespace std;

int main() {
   MPI_Init(NULL, NULL); 

   int CFD_realm = 1, CFD_COMM;
   CPL::init(CFD_realm, CFD_COMM);

   int nprocs; int rank;
   MPI_Comm_size(CFD_COMM, &nprocs);   
   MPI_Comm_rank(CFD_COMM, &rank);

   cout << "MD code processor " << rank+1 << " of " << nprocs << endl;

   CPL::finalize();
   MPI_Finalize();
}



